src/write_grpc.c

```diff
@@ -810,80 +819,19 @@ static int write_data(const data_set_t *ds, const value_list_t *vl,
  * from the GCE developer's console.  Returns JWT credentials (bound
  * to use only the gRPC channel service).
  */
-static grpc_credentials *get_client_credentials(
-    bool use_instance_credentials,
-    const char *service_account_json_filename)
-{
-  grpc_credentials *cred_client = NULL;
-
-  if (use_instance_credentials) {
-    cred_client = grpc_compute_engine_credentials_create();
-    if (cred_client != NULL) {
-      INFO("Loaded GCE credentials from instance metadata server");
-      if (service_account_json_filename != NULL) {
-        WARNING("Already have credentials; ignoring ServiceAccountJsonFile %s",
-                service_account_json_filename);
-      }
-      return cred_client;
-    }
-    /* Keep going: might succeed with a configured service account */
-    ERROR("Failed to fetch GCE credentials from instance metadata server");
-  }
-
-  if (service_account_json_filename != NULL) {
-    char json[8192];
-    ssize_t json_len;
-
-    if ((json_len = read_file_contents(
-            service_account_json_filename, json, sizeof(json) - 1)) < 0) {
-      ERROR("Failed to read ServiceAccountJsonFile %s",
-            service_account_json_filename);
-      return NULL;
-    }
-    json[json_len] = '\0';  /* Ensure readable string */
-    if (! (cred_client = grpc_jwt_credentials_create(
-            json, grpc_max_auth_token_lifetime))) {
-      ERROR("Failed to create credentials from ServiceAccountJsonFile %s",
-            service_account_json_filename);
-      return NULL;
-    }
-    return cred_client;
-  }
-
-  ERROR("Must specify UseInstanceCredentials or ServiceAccountJsonFile");
-  return NULL;
-}
-
-/* Returns cred bound with root CA for authenticating the server */
-static grpc_credentials *get_server_credentials(const char *root_pem_filename)
-{
-  grpc_credentials *cred_server_ssl = NULL;
-
-  // TODO(arielshaqed): Do we need this? Why not just get user to setenv?
-  if (root_pem_filename)
-    setenv("GRPC_DEFAULT_SSL_ROOTS_FILE_PATH", root_pem_filename, 1);
-  cred_server_ssl = grpc_ssl_credentials_create(NULL, NULL);
-  return cred_server_ssl;
-}
-
 /* Returns credentials for an authenticated connection */
 static grpc_credentials *get_credentials(
     bool use_instance_credentials,
     const char *service_account_json_filename,
     const char *root_pem_filename)
 {
-  grpc_credentials *cred_client = get_client_credentials(
-      use_instance_credentials, service_account_json_filename);
-  grpc_credentials *cred_server = get_server_credentials(root_pem_filename);
-  grpc_credentials *ret = NULL;
-
-  if (cred_client && cred_server)
-    ret = grpc_composite_credentials_create(cred_server, cred_client);
-
-  if (cred_client) grpc_credentials_release(cred_client);
-  if (cred_server) grpc_credentials_release(cred_server);
+  if (service_account_json_filename != NULL)
```

Comment:
Environment variables are the only way to pass these parameters? Specifically for GRPC_GOOGLE_CREDENTIALS_ENV_VAR, if it's not set, will it go through the metadata server?